Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -375,6 +376,18 @@ protected virtual Page CreatePageFromSegment(string segment)
             }
         }
 
+        protected void ApplyPageBehaviors(Page page)
```

Comment:
If we make this virtual that will allow developers to easily add other behaviors to existing page types or handle any events for a custom page type.